comm.go

```diff
@@ -172,19 +172,15 @@ func (p *PubSub) handleSendingMessages(ctx context.Context, s network.Stream, ou
 
 	defer s.Close()
 	for {
-		select {
-		case rpc, ok := <-outgoing:
-			if !ok {
-				return
-			}
+		rpc, err := outgoing.Pop(ctx)
+		if err != nil {
+			return
```

Comment:
We might want to log the error (at very low priority) if it is not a closed indicator.